src/controllers/connection/__tests__/helpers.ts

```diff
@@ -103,23 +116,46 @@ export const withConnectionMocks = (
 }
 
 export const withNewConnectionMocks = () => {
-  const mockTransactionDb = {
+  const mockWithTransaction = {
     insert: () => Promise.resolve([{ id: '42' }]),
     get: () => Promise.resolve([validConnection]),
     update: () => Promise.resolve(),
   }
   const mockDb = {
     get: (tableName: string, where?: Record<string, string>) => {
-      if (tableName !== 'connection') throw new Error('Invalid table')
-      if (where?.company_number === validCompanyNumber) return []
-      if (where?.company_number === validExistingCompanyNumber) return [{}]
-      if (where?.id === '4a5d4085-5924-43c6-b60d-754440332e3d') return [validConnection]
-      return []
+      if (tableName === 'connection') {
+        if (where?.company_number === validCompanyNumber) return []
+        if (where?.id === '4a5d4085-5924-43c6-b60d-754440332e3d') return [validConnection]
+        // sending a second invite test setup - sad path cases
+        if (where?.company_number === noExistingInviteCompanyNumber) return [{}]
+        if (where?.company_number === verifiedBothCompanyNumber) return [{ status: 'verified_both' }]
+        if (where?.company_number === usedPendingCompanyNumber) return [{ id: usedInvite, status: 'pending' }]
+        if (where?.company_number === usedVerifiedThemCompanyNumber)
+          return [{ id: usedInvite, status: 'verified_them' }]
+        if (where?.company_number === usedDisconnectedCompanyNumber) return [{ id: usedInvite, status: 'disconnected' }]
+        if (where?.company_number === tooManyDisconnectedCompanyNumber)
+          return [{ id: tooManyInvite, status: 'disconnected' }]
+        if (where?.company_number === validDisconnectedCompanyNumber)
+          return [{ id: validInvite, status: 'disconnected' }]
+        // happy path cases for sending a second invite
+        if (where?.company_number === usedUnverifiedCompanyNumber) return [{ id: usedInvite, status: 'unverified' }] // happy path
+        if (where?.company_number === usedVerifiedUsCompanyNumber) return [{ id: usedInvite, status: 'verified_us' }] // happy path
+        if (where?.company_number === validPendingCompanyNumber) return [{ id: validInvite, status: 'pending' }]
+        return []
```

Comment:
there might be a way of rewriting this with a map `company_number: connection`. Like 
```
{
verifiedBothCompanyNumber: {'id': usedInvite, status: 'verified_both'},
...
}

then these lines can be replaced by `return companyNumberToConnectionMap[companyNumber]` and it's easier to read and edit

Reply:
Thanks! Anything to make it easier :)

Reply:
Done